pkg/protocol/engine/congestioncontrol/scheduler/drr/scheduler.go

```diff
@@ -337,15 +328,66 @@ func (s *Scheduler) selectIssuer(start *IssuerQueue) (int64, *IssuerQueue) {
 	return rounds, schedulingIssuer
 }
 
+func (s *Scheduler) removeIssuer(q *IssuerQueue) {
+	q.submitted.ForEach(func(id iotago.BlockID, block *blocks.Block) bool {
+		block.SetDropped()
+		s.events.BlockDropped.Trigger(block)
+
+		return true
+	})
+
+	for q.inbox.Len() > 0 {
+		block := q.PopFront()
```

Comment:
Could also be racy, what if the inbox had an element during the length check, but not during PopFront?

Reply:
Protected from outside by bufferMutex.